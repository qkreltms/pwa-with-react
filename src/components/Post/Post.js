import React from 'react'
import styles from './Post.css'
import { Link } from 'react-router-dom'

const Post = (props) => {
  return (
    <div>
      <h2 className={styles.title}>
        <Link className={styles.link} to={`/post/${props.slug}`}>{props.title}</Link></h2>
      <p className={styles.content}>{props.excerpt}</p>
    </div>
  )
}

export default Post
