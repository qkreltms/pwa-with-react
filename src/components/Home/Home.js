import React from 'react'
import Post from '../Post/Post'

const Home = ({ posts }) => {
  return (
    posts.length
      ? posts.map(post => <Post {...post} key={post.slug} />)
      : <p> Please Wait...</p>
  )
}

export default Home
