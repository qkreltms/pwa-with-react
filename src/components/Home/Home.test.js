/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import Home from './Home'
import toJson from 'enzyme-to-json'
import Post from '../Post/Post'

test('About component should render as expected', () => {
  // 스냅 데이터로 test
  const posts = [{
    'title': 't1',
    'slug': 's1',
    'excerpt': 'e1',
    'content': 'c1'
  }]
  const component = shallow(<Home posts={posts} />)
  const tree = toJson(component)

  expect(component.find(Post).length).toBe(1)
  expect(tree).toMatchSnapshot()
})

test('About component should render as expected', () => {
  const component = shallow(<Home posts={[]} />)
  const tree = toJson(component)

  expect(component.find(Post).length).toBe(0)
  expect(tree).toMatchSnapshot()
})
