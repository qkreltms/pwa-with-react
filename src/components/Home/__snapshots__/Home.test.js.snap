// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component should render as expected 1`] = `
<Post
  content="c1"
  excerpt="e1"
  key="s1"
  slug="s1"
  title="t1"
/>
`;

exports[`About component should render as expected 2`] = `
<p>
   Please Wait...
</p>
`;
